service-base/src/main/java/org/eclipse/hono/service/tenant/ExecutionContextTenantAndAuthIdProvider.java

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.tenant;
+
+import org.eclipse.hono.util.ExecutionContext;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+
+/**
+ * Provides a method to determine the tenant and auth-id of a protocol adapter request from the given ExecutionContext.
+ *
+ * @param <T> The type of ExecutionContext used.
+ */
+public interface ExecutionContextTenantAndAuthIdProvider<T extends ExecutionContext> {
+
+    /**
+     * Get the tenant and auth-id from the given ExecutionContext.
+     *
+     * @param context The execution context.
+     * @param spanContext The OpenTracing context to use for tracking the operation.
```

Comment:
is this optional?

Reply:
Yes, I've changed the javadoc accordingly.